src/v2/view-builder/views/email/ChallengeAuthenticatorEmailView.js

```diff
@@ -61,15 +68,24 @@ const Body = BaseAuthenticatorEmailForm.extend(
       const { email } =
         this.options.currentViewState.relatesTo?.value?.profile || {};
 
-      this.add(EnterCodeLink, {
-        prepend: true,
-        selector: '.o-form-error-container',
-      });
+      const useEmailMagicLink = this.options.appState.get('currentAuthenticatorEnrollment')?.
+      contextualData?.useEmailMagicLink;
+      const useEmailMagicLinkValue = useEmailMagicLink !== undefined ? useEmailMagicLink : true;
+
+      //If email magic link exist then show button with text, else toggle button to show text box
+      if (useEmailMagicLinkValue) {
```

Comment:
I think you can just use if (!!useEmailMagicLink) and line 73 is not necessary,

Reply:
!! -> does this check for undefined and true both?

Reply:
I think so,
` value     │  !value  │  !!value
━━━━━━━━━━━┿━━━━━━━━━━┿━━━━━━━━━━━
 false     │ ✔ true   │   false
 true      │   false  │ ✔ true
 null      │ ✔ true   │   false
 undefined │ ✔ true   │   false
 0         │ ✔ true   │   false
 -0        │ ✔ true   │   false
 1         │   false  │ ✔ true
 -5        │   false  │ ✔ true
 NaN       │ ✔ true   │   false
 ''        │ ✔ true   │   false
 'hello'   │   false  │ ✔ true`

Reply:
I think so
![Screen Shot 2022-02-01 at 5 18 28 PM](https://user-images.githubusercontent.com/87032825/152061062-979a457a-0af0-4b26-b3a8-e0979b8a56b1.png)

Reply:
this wont work for me because if you see the logic I wrote, when its undefined I want it as true. But as per chart !! returns false.